src/platforms/esp32/components/avm_sys/sys.c

```diff
@@ -499,13 +499,13 @@ static term get_features(Context *ctx, uint32_t features)
 term sys_get_info(Context *ctx, term key)
 {
     GlobalContext *glb = ctx->global;
-    if (key == globalcontext_make_atom(glb, esp_free_heap_size_atom)) {
+    if (key == globalcontext_make_atom(glb, atomvm_free_heap_size_atom)) {
         return term_from_int32(esp_get_free_heap_size());
     }
     if (key == globalcontext_make_atom(glb, esp_largest_free_block_atom)) {
         return term_from_int32(heap_caps_get_largest_free_block(MALLOC_CAP_DEFAULT));
     }
-    if (key == globalcontext_make_atom(glb, esp_get_minimum_free_size_atom)) {
+    if (key == globalcontext_make_atom(glb, atomvm_get_minimum_free_size_atom)) {
```

Comment:
We should be more gradual, we are in v0.x stage and we can do that, but it is not the best experience for our users.
I suggest:  `((key == globalcontext_make_atom(glb, atomvm_get_minimum_free_size_atom) || (key == globalcontext_make_atom(glb, esp_get_minimum_free_size_atom))` + adding a TODO about the plan to remove this + mentioning this change in UPDATING.md + adding "Breaking: ` to the relevant changelog lines.

Reply:
This makes sense, in fact I think we may want to print a warning when the older keys are used and inform users to update to the new keys before they are removed in a future release.

Reply:
Let's also add a TODO on each branch we plan to remove, so we can find them easier.

Reply:
That sounds like a good plan. Should they be removed in the 0.8 release cycle? This will give uses the 0.7 release to see the warning and update their applications.